segmentation/utils/spatial_analysis_utils_test.py

```diff
@@ -0,0 +1,308 @@
+    assert (example_closenum[3:5, 3:5] == 25).all()
+    assert (example_closenum[5:7, 5:7] == 1).all()
+
+    # Now test with Erin's output
+    # cell_array = pd.read_csv(
+    #     "/Users/jaiveersingh/Downloads/SpatialEnrichm"
+    #     "ent/granA_cellpheno_CS-asinh-norm_revised.csv")
+    # marker_thresholds = pd.read_csv(
+    #     "/Users/jaiveersingh/Downloads/SpatialEnrichment/markerThresholds.csv")
+    # dist_matrix = np.asarray(pd.read_csv(
+    #     "/Users/jaiveersingh/Documents/MATLAB/distancesMat5.csv", header=None))
+    # closenum, closenumRand, z, muhat, sigmahat, p, h, adj_p, markertitles = \
+    #     spatialorgmarkerexpression_utils.calculate_channel_spatial_enrichment(
+    #         dist_matrix, marker_thresholds, cell_array)
+    # real_closenum = np.asarray(pd.read_csv(
+    #     "/Users/jaiveersingh/Documents/MATLAB/SpatialAnalysis/closeNum.csv"))
+    # assert np.array_equal(closenum, real_closenum)
+
+
+def test_compute_close_cell_num_random():
+    example_all_patient_data, example_distmat = make_example_data_closenum()
+
+    # Generate random inputs to test shape
+    marker1_num = [random.randrange(0, 10) for i in range(20)]
+    marker2_num = [random.randrange(0, 5) for i in range(400)]
+    marker_num = 20
+
+    example_closenumrand = spatial_analysis.compute_close_cell_num_random(
+        marker1_num, marker2_num, example_distmat, marker_num, dist_lim=100,
+        bootstrap_num=100)
+
+    assert example_closenumrand.shape == (20, 20, 100)
+
+
+def test_calculate_enrichment_stats():
+    # Positive enrichment
+
+    # Generate random closenum matrix
+    stats_cnp = np.zeros((20, 20))
+    stats_cnp[:, :] = 80
+
+    # Generate random closenumrand matrix, ensuring significant positive enrichment
+    stats_cnrp = np.random.randint(1, 40, (20, 20, 100))
+
+    stats_xr_pos = spatial_analysis.calculate_enrichment_stats(stats_cnp, stats_cnrp)
+
+    assert stats_xr_pos.loc["z", 0, 0] > 0
+    assert stats_xr_pos.loc["p_pos", 0, 0] < .05
+
+    # Negative enrichment
+
+    # Generate random closenum matrix
+    stats_cnn = np.zeros((20, 20))
+
+    # Generate random closenumrand matrix, ensuring significant negative enrichment
+    stats_cnrn = np.random.randint(40, 80, (20, 20, 100))
+
+    stats_xr_neg = spatial_analysis.calculate_enrichment_stats(stats_cnn, stats_cnrn)
+
+    assert stats_xr_neg.loc["z", 0, 0] < 0
+    assert stats_xr_neg.loc["p_neg", 0, 0] < .05
+
+    # No enrichment
+
+    # Generate random closenum matrix
+    stats_cn = np.zeros((20, 20))
+    stats_cn[:, :] = 80
+
+    # Generate random closenumrand matrix, ensuring no enrichment
+    stats_cnr = np.random.randint(78, 82, (20, 20, 100))
+
+    stats_xr = spatial_analysis.calculate_enrichment_stats(stats_cn, stats_cnr)
+
+    assert abs(stats_xr.loc["z", 0, 0]) < 1
+    assert stats_xr.loc["p_neg", 0, 0] > .05
+    assert stats_xr.loc["p_pos", 0, 0] > .05
+
+
+def test_calculate_channel_spatial_enrichment():
```

Comment:
Let's move this to function to a new file, called spatial_analysis_test, since it's testing a function in spatial_analysis, not spatial_analysis_utils